core/lib/src/serde/msgpack.rs

```diff
@@ -123,9 +113,61 @@ pub use rmp_serde::decode::Error;
 /// msgpack = 5242880
 /// ```
 #[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
-pub struct MsgPack<T>(pub T);
+pub struct MsgPack<T, const COMPACT: bool = true>(pub T);
```

Comment:
I think we should change the default here. This is likely to be the more common case, and provides better interoperability with potential client code.